src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5545,6 +5545,25 @@ static QVariant fcnHashSha256( const QVariantList &values, const QgsExpressionCo
   return fcnHash( QgsExpressionUtils::getStringValue( values.at( 0 ), parent ), QCryptographicHash::Sha256 );
 }
 
+static QVariant fcnToBase64( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QString value = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
+
+  const QByteArray input = value.toLocal8Bit();
+
+  return QVariant( QString( input.toBase64() ) );
+}
+
+static QVariant fcnFromBase64( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QString value = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
+
+  const QByteArray base64 = value.toLocal8Bit();
+  const QByteArray decoded = QByteArray::fromBase64( base64 );
+
+  return QVariant( QString( decoded ) );
```

Comment:
I don't think this should be forced to a QString here and preserved as raw bytes.

Reply:
And this one should be:

     const QString value = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
     const QByteArray base64 = value.toLocal8Bit();
     const QByteArray decoded = QByteArray::fromBase64( base64 );
     return QVariant( decoded );